src/System.Management.Automation/engine/Utils.cs

```diff
@@ -1255,11 +1256,24 @@ internal static bool IsReservedDeviceName(string destinationPath)
             return false;
         }
 
+        private static readonly Regex _specialDriveRegex = new Regex(@"^\\\\\w+\$");
+
         internal static bool PathIsUnc(string path)
         {
 #if UNIX
             return false;
 #else
+            if (!path.StartsWith('\\'))
+            {
+                return false;
+            }
+
+            // handle special cases like \\wsl$\ubuntu which isn't a UNC path, but we can say it is so the filesystemprovider can use it
+            if (_specialDriveRegex.IsMatch(path))
```

Comment:
I found docs saying that it is valid UNC path! :-) https://docs.microsoft.com/en-us/openspecs/windows_protocols/ms-dtyp/62e862f4-2a51-452e-8eeb-dc4ff5ee33cc
>  pchar              = %x20-21 / %x23-29

where x24 is '$'.

But I very wonder why we should add slow Regex if we need have only workaround for the specific path `\\wsl$\`. I suggest string.StartWith(@"\\wsl$\", ...)

Reply:
I didn't want to hard code it to `wsl$` and I don't know if in the future other special paths would exist.  Even though it may be a valid UNC path, the System.Uri class `IsUnc` property says it's not.  Perhaps we can optimize it for `wsl$` for now until we get reports of issues with other paths.